src/main/java/core/basesyntax/service/operations/ReturnOperation.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax.service.operations;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.exceptions.InvalidOperationException;
+
+public class ReturnOperation implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
+        if (transaction.getOperation() != FruitTransaction.Operation.RETURN) {
+            throw new InvalidOperationException("Invalid operation. Should be: 'RETURN', but was: "
+                    + transaction.getOperation() + ".");
+        }
+        if (!Storage.containsKey(transaction.getFruit())) {
+            throw new RuntimeException("Item: '" + transaction.getFruit()
+            + "' doesn't exist in the storage");

```

Comment:
Instead of throwing a generic 'RuntimeException', consider creating a more specific exception. This will make the error handling more precise and easier to understand.

Reply:
ok, done